src/Confluent.Kafka/Exceptions/MessageNullException.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Represents an error that occures when <c>ConsumeResult</c> returns with a partition EOF signal. The message will be Null in this case and the convenience properties cannot be used.
```

Comment:
I think too specific. Maybe: 'Thrown when there is an attempt to dereference a null Message reference.'